services/uinotification/notify.go

```diff
@@ -43,66 +54,79 @@ var _ notify_service.Notifier = &notificationService{}
 // NewNotifier create a new notificationService notifier
 func NewNotifier() notify_service.Notifier {
 	ns := &notificationService{}
-	ns.issueQueue = queue.CreateSimpleQueue(graceful.GetManager().ShutdownContext(), "notification-service", handler)
-	if ns.issueQueue == nil {
+	ns.queue = queue.CreateSimpleQueue(graceful.GetManager().ShutdownContext(), "notification-service", handler)
+	if ns.queue == nil {
 		log.Fatal("Unable to create notification-service queue")
 	}
 	return ns
 }
 
-func handler(items ...issueNotificationOpts) []issueNotificationOpts {
+func handler(items ...notificationOpts) []notificationOpts {
 	for _, opts := range items {
-		if err := activities_model.CreateOrUpdateIssueNotifications(db.DefaultContext, opts.IssueID, opts.CommentID, opts.NotificationAuthorID, opts.ReceiverID); err != nil {
-			log.Error("Was unable to create issue notification: %v", err)
+		switch opts.Source {
+		case activities_model.NotificationSourceRepository:
+			if err := activities_model.CreateRepoTransferNotification(db.DefaultContext, opts.NotificationAuthorID, opts.RepoID, opts.ReceiverID); err != nil {
+				log.Error("CreateRepoTransferNotification: %v", err)
+			}
+		case activities_model.NotificationSourceCommit:
+			if err := activities_model.CreateCommitNotifications(db.DefaultContext, opts.NotificationAuthorID, opts.RepoID, opts.CommitID, opts.ReceiverID); err != nil {
+				log.Error("Was unable to create commit notification: %v", err)
+			}
+		case activities_model.NotificationSourceRelease:
+			if err := activities_model.CreateOrUpdateReleaseNotifications(db.DefaultContext, opts.NotificationAuthorID, opts.RepoID, opts.ReleaseID, opts.ReceiverID); err != nil {
+				log.Error("Was unable to create release notification: %v", err)
+			}
+		case activities_model.NotificationSourceIssue, activities_model.NotificationSourcePullRequest:
+			fallthrough
+		default:
+			if err := activities_model.CreateOrUpdateIssueNotifications(db.DefaultContext, opts.IssueID, opts.CommentID, opts.NotificationAuthorID, opts.ReceiverID); err != nil {
+				log.Error("Was unable to create issue notification: %v", err)
+			}
```

Comment:
Should it panic in dev mode to warn developers that there is a mistake if the "opt" is not for "issue"?